src/ValueParsers/CompositeValueParser.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+namespace ValueParsers;
+
+use InvalidArgumentException;
+
+/**
+ * A generic value parser that forwards parsing to a list of other value parsers and returns the
+ * result of the first parse attempt that succeeded.
+ *
+ * @since 0.5
+ *
+ * @licence GNU GPL v2+
+ * @author Thiemo Mättig
+ */
+class CompositeValueParser implements ValueParser {
+
+	/**
+	 * @var ValueParser[]
+	 */
+	private $parsers;
+
+	/**
+	 * @see ParseException::getExpectedFormat
+	 *
+	 * @var string
+	 */
+	private $format;
+
+	/**
+	 * @param ValueParser[] $parsers
+	 * @param string $format An identifier describing the expected format of the values to parse.
+	 *
+	 * @throws InvalidArgumentException
+	 */
+	public function __construct( array $parsers, $format ) {
+		if ( empty( $parsers ) ) {
+			throw new InvalidArgumentException( '$parsers must be a non-empty array' );
+		}
+
+		if ( !is_string( $format ) || $format === '' ) {
+			throw new InvalidArgumentException( '$format must be a non-empty string' );
+		}
+
+		$this->parsers = $parsers;
+		$this->format = $format;
+	}
+
+	/**
+	 * @param mixed $value
+	 *
+	 * @throws ParseException
+	 * @return mixed
+	 */
+	public function parse( $value ) {
+		foreach ( $this->parsers as $parser ) {
+			try {
+				return $parser->parse( $value );
+			} catch ( ParseException $ex ) {
+				continue;
```

Comment:
This is a rather icky. Generally you do not want to use exceptions for expected execution paths, which is happening here.